execute/transformation.go

```diff
@@ -9,11 +9,17 @@ import (
 	"github.com/influxdata/flux/plan"
 )
 
+// Transformation represents functions that control of how the execution
+// engine will take in a flux Table, perform the necessary data processing
+// and return an output flux Table
 type Transformation interface {
 	RetractTable(id DatasetID, key flux.GroupKey) error
+	// Process takes in a flux Table, performs data processing on it
+	// and produces an output flux Table
 	Process(id DatasetID, tbl flux.Table) error
 	UpdateWatermark(id DatasetID, t Time) error
 	UpdateProcessingTime(id DatasetID, t Time) error
+	// Finish indicates that the Transformation is complete
 	Finish(id DatasetID, err error)
```

Comment:
What does `complete` mean? I can answer that one, `finished` or  `complete` mean that no other methods on the Transformation will be called. I.e. its always the last method called on the transformation.

Reply:
That's fair. I can update this to be more clear.